P0009/P0009r12.md

```diff
@@ -2202,21 +2282,21 @@ define the values of `first[r]` and `last[r]` as follows:
 
 ```c++
 // Create a mapping
-typedef extents<3,dynamic_extent,7> Extents3D;
+using Extents3D = extents<3, dynamic_extent, 7>;
 layout_right::template mapping<Extents3D> map_right(10);
 
-// Allocate a basic_mdspan
+// Create an mdspan viewing allocated memory
 int* ptr = new int[3*8*10];
-basic_mdspan<int,Extents3D,layout_right> a(ptr,map_right);
+mdspan<int, Extents3D, layout_right> a(ptr, map_right);
 
 // Initialize the span
 for(int i0=0; i0<a.extent(0); i0++)
   for(int i1=0; i1<a.extent(1); i1++)
     for(int i2=0; i2<a.extent(2); i2++)
-      a(i0,i1,i2) = 10000*i0+100*i1+i2;
+      a(i0,i1,i2) = 10000*i0 + 100*i1 + i2;
 
 // Create Subspan
-auto a_sub = submdspan(a,1,pair<int,int>(4,6),pair<int,int>(1,6));
+auto a_sub = submdspan(a, 1, pair<int,int>(4,6), pair<int,int>(1,6));
```

Comment:
Now we have CTAD, could we simplify the code, like
```suggestion
auto a_sub = submdspan(a, 1, pair { 4, 6 }, pair { 1, 6 });
```
?
Could it be possible to reach
```c++
auto a_sub = submdspan(a, 1, { 4, 6 }, { 1, 6 });
```
?

Reply:
[CTAD works fine](https://godbolt.org/z/rvbTcqT8e), but unadorned curly braces do not work.

I'll be happy to make this change, but please remember that examples are nonnormative.

Reply:
hi, one of the goals was to leave this C++14 compliant for now. Largely due to limitations in some of the customer software stacks (like LLNL Sierra, the top 3 HPC machine in the world can't fully use C++17 yet ...).